core/src/test/scala/org/apache/spark/util/collection/FixedHashObject.scala

```diff
@@ -22,4 +22,8 @@ package org.apache.spark.util.collection
  */
 case class FixedHashObject(v: Int, h: Int) extends Serializable {
   override def hashCode(): Int = h
+  override def equals(other: Any): Boolean = other match {
```

Comment:
Although this very technically changes semantics, given the clear and specific purpose of this class, I don't think it hurts to define reasonable equality semantics instead of leaving it to reference equality.